cmd/configure/main.go

```diff
@@ -1,89 +1,21 @@
 package main
 
 import (
-	"errors"
-	"fmt"
-	"io/ioutil"
 	"log"
-	"os"
-	"path/filepath"
-	"regexp"
-	"text/template"
+
+	"github.com/paketo-buildpacks/nginx/cmd/configure/internal"
 )
 
 func main() {
 	log.SetFlags(0)
 
-	mainConf := os.Args[1]
-	body, err := ioutil.ReadFile(mainConf)
-	if err != nil {
-		log.Fatalf("Could not read config file (%s): %s", mainConf, err)
-	}
+	err := internal.Run(
+		"/workspace/nginx.conf",
+		"/workspace/modules",
+		"/layers/paketo-buildpacks_nginx/nginx/modules",
```

Comment:
This path can be reverse engineered from `os.Args[0]` which is the path of the running executable, `configure` in this case. This path points at a location inside the layers directory where this executable resides, `/layers/paketo-buildpacks_nginx/nginx/exec.d/0-configure`. You can take that path and trim off the `exec.d/0-configure` bits to get the layer path which you can then append `modules` onto.

Reply:
@ryanmoran Good point!